estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,51 @@
+from odoo import models, fields, api, _
+from datetime import timedelta
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Property offer"
+    _order = "price desc"
+
+    price = fields.Float(required=True)
+    status = fields.Selection(selection=[("accepted", "Accepted"), ("refused", "Refused")])
+    partner_id = fields.Many2one("res.partner", string="Partner", required=True)
+    property_id = fields.Many2one("estate.property", string="Property", required=True)
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+    property_type_id = fields.Many2one(related="property_id.property_type_id", store=True)
+
+    _sql_constraints = [
+        ('check_price_positive', 'CHECK(price >= 0)',
+            'The offer price of should be greater than 0.'),
+    ]
+
+    @api.depends("validity", "create_date")
+    def _compute_date_deadline(self):
+        for record in self:
+            if record.create_date:
+                record.date_deadline = record.create_date + timedelta(days=record.validity)
+            else:
+                record.date_deadline = fields.Date.today() + timedelta(days=record.validity)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            if record.date_deadline and record.create_date:
+                delta = record.date_deadline - record.create_date.date()
+                record.validity = delta.days
+            else:
+                record.validity = (record.date_deadline - fields.Date.today()).days
+
+    def action_accept(self):
+        for record in self:
+            if record.property_id.estate == "sold":
+                raise UserError(_("Cannot accept off  er for a sold property"))
+            record.status = "accepted"
+            record.property_id.buyer_id = record.partner_id
+            record.property_id.estate = "offer accepted"
```

Comment:
Did you test this? I don't see any reference for this field in the code

this will crash the server, i think it was changed for state